Added unknown type for rsrc

ITAYC0HEN opened the pull request:
Changed the `return "UNKNOWN"` to `"UNKNOWN (%d)", type`.
Now, whenever the resource type is UNKNOWN, there will still be the type in parentheses.

radare commented on libr/bin/format/pe/pe.c:
```diff
@@ -2189,28 +2189,28 @@ static char* _resource_lang_str(int id) {
-	case 23: return "HTML";
-	case 24: return "MANIFEST";
-	default: return "UNKNOWN";
+	case 1: return strdup("CURSOR");
```
add space before (

radare commented on libr/bin/format/pe/pe.c:
```diff
@@ -2189,28 +2189,28 @@ static char* _resource_lang_str(int id) {
-	case 24: return "MANIFEST";
-	default: return "UNKNOWN";
+	case 1: return strdup("CURSOR");
+	case 2: return strdup("BITMAP");
```
better assign a local variable and strdup once, instead of having 30 strdups

radare commented on libr/bin/format/pe/pe.c:
```diff
@@ -2188,29 +2188,53 @@ static char* _resource_lang_str(int id) {
-	case 23: return "HTML";
-	case 24: return "MANIFEST";
-	default: return "UNKNOWN";
+		case 1: typeName = "CURSOR";
```
we dont indent cases and we dont put stuff at the right of the :, put it below or just an array of strings and just one if to assign the Nth element of the string or return r_str_newf, that will reduce this function to 5-10 lines

ITAYC0HEN commented:
An array of strings is a good idea, although not-so-clear and readable (like, to map TypeName to TypeNum in the eyes).

Also, the cases look indented here, with two spaces :( 
https://github.com/radare/radare2/blob/master/DEVELOPERS.md#code-style

radare commented:
Please find the latest report on new defect(s) introduced to radare2 found with Coverity Scan.

5 new defect(s) introduced to radare2 found with Coverity Scan.
18 defect(s), reported by Coverity Scan earlier, were marked fixed in the recent build analyzed by Coverity Scan.

New defect(s) Reported-by: Coverity Scan
Showing 5 of 5 defect(s)


** CID 1393580:  Memory - corruptions  (OVERRUN)
/libr/core/core.c: 962 in autocomplete_default()


________________________________________________________________________________________________________
*** CID 1393580:  Memory - corruptions  (OVERRUN)
/libr/core/core.c: 962 in autocomplete_default()
956                 int length = strlen (radare_argv[i]);
957                 if (!strncmp (radare_argv[i], line->buffer.data, length)) {
958                     tmp_argv[j++] = radare_argv[i];
959                 }
960             }
961         }
   CID 1393580:  Memory - corruptions  (OVERRUN)
   Overrunning array "tmp_argv" of 512 8-byte elements at element index 512 (byte offset 4096) using index "j" (which evaluates to 512).
962         tmp_argv[j] = NULL;
963         line->completion.argc = j;
964         line->completion.argv = tmp_argv;
965     }
966     
967     static void autocomplete_evals(RLine* line, const char* input) {

** CID 1393579:  Code maintainability issues  (SIZEOF_MISMATCH)
/libr/core/core.c: 3110 in r_core_autocomplete_remove()


________________________________________________________________________________________________________
*** CID 1393579:  Code maintainability issues  (SIZEOF_MISMATCH)
/libr/core/core.c: 3110 in r_core_autocomplete_remove()
3104             if (r_str_glob (ac->cmd, cmd)) {
3105                 for (j = i + 1; j < parent->n_subcmds; ++j) {
3106                     parent->subcmds[j - 1] = parent->subcmds[j];
3107                     parent->subcmds[j] = NULL;
3108                 }
3109                 r_core_autocomplete_free (ac);
   CID 1393579:  Code maintainability issues  (SIZEOF_MISMATCH)
   Passing argument "parent->subcmds" of type "RCoreAutocomplete **" and argument "(parent->n_subcmds - 1) * 8UL /* sizeof (RCoreAutocomplete **) */" to function "realloc" is suspicious. In this case, "sizeof (RCoreAutocomplete **)" is equal to "sizeof (RCoreAutocomplete *)", but this is not a portable assumption.
3110                 RCoreAutocomplete **updated = realloc (parent->subcmds, (parent->n_subcmds - 1) * sizeof (RCoreAutocomplete**));
3111                 if (!updated && (parent->n_subcmds - 1) > 0) {
3112                     eprintf ("Something really bad has happen.. this should never ever happen..\n");
3113                     return false;
3114                 }
3115                 parent->subcmds = updated;

** CID 1393578:  Null pointer dereferences  (REVERSE_INULL)
/libr/core/core.c: 1156 in autocomplete_macro()


________________________________________________________________________________________________________
*** CID 1393578:  Null pointer dereferences  (REVERSE_INULL)
/libr/core/core.c: 1156 in autocomplete_macro()
1150         RListIter *iter;
1151         char buf[1024];
1152         int n, i = 0;
1153         n = strlen(str);
1154         r_list_foreach (core->rcmd->macro.macros, iter, item) {
1155             char *p = item->name;
   CID 1393578:  Null pointer dereferences  (REVERSE_INULL)
   Null-checking "str" suggests that it may be null, but it has already been dereferenced on all paths leading to the check.
1156             if (!str || !*str || !strncmp (str, p, n)) {
1157                 snprintf (buf, sizeof (buf), "%s%s)", str, p);
1158                 // eprintf ("------ %p (%s) = %s\n", tmp_argv[i], buf, p);
1159                 if (r_is_heap ((void*)tmp_argv[i])) {
1160                     free ((char *)tmp_argv[i]);
1161                 }

** CID 1393577:  Memory - corruptions  (OVERRUN)
/libr/core/core.c: 1296 in find_autocomplete()


________________________________________________________________________________________________________
*** CID 1393577:  Memory - corruptions  (OVERRUN)
/libr/core/core.c: 1296 in find_autocomplete()
1290             int length = strlen (arg);
1291             for (i = 0; j < TMP_ARGV_SZ && i < parent->n_subcmds; i++) {
1292                 if (!strncmp (arg, parent->subcmds[i]->cmd, length)) {
1293                     tmp_argv[j++] = parent->subcmds[i]->cmd;
1294                 }
1295             }
   CID 1393577:  Memory - corruptions  (OVERRUN)
   Overrunning array "tmp_argv" of 512 8-byte elements at element index 512 (byte offset 4096) using index "j" (which evaluates to 512).
1296             tmp_argv[j] = NULL;
1297             line->completion.argc = j;
1298             line->completion.argv = tmp_argv;
1299             break;
1300         }
1301         return true;

** CID 1393581:  Code maintainability issues  (SIZEOF_MISMATCH)
/libr/core/core.c: 3049 in r_core_autocomplete_add()


________________________________________________________________________________________________________
*** CID 1393581:  Code maintainability issues  (SIZEOF_MISMATCH)
/libr/core/core.c: 3049 in r_core_autocomplete_add()
3043             return NULL;
3044         }
3045         RCoreAutocomplete *autocmpl = R_NEW0 (RCoreAutocomplete);
3046         if (!autocmpl) {
3047             return NULL;
3048         }
   CID 1393581:  Code maintainability issues  (SIZEOF_MISMATCH)
   Passing argument "parent->subcmds" of type "RCoreAutocomplete **" and argument "(parent->n_subcmds + 1) * 8UL /* sizeof (RCoreAutocomplete **) */" to function "realloc" is suspicious. In this case, "sizeof (RCoreAutocomplete **)" is equal to "sizeof (RCoreAutocomplete *)", but this is not a portable assumption.
3049         RCoreAutocomplete **updated = realloc (parent->subcmds, (parent->n_subcmds + 1) * sizeof(RCoreAutocomplete**));
3050         if (!updated) {
3051             free (autocmpl);
3052             return NULL;
3053         }
3054         parent->subcmds = updated;
Ho

radare commented:
Fuck clipboard

radare commented:
Who the hell used spaces in that .md

The cases must be at the same level as the switch and the body of them must gi below

ITAYC0HEN commented:
I fixed DEVELOPERS.md -- sent different PR.

Regarding the array of strings -- surprisingly, there's no full sequence of resource types (for example, types 13,14,15 do not exist according to [Microsoft](https://msdn.microsoft.com/en-us/library/windows/desktop/ms648009(v=vs.85).aspx)) so using an array and indexes isn't possible.
I fixed the indention of the switch cases.
